src/transformers/models/idefics/modeling_idefics.py

```diff
@@ -980,16 +980,46 @@ def _init_weights(self, module):
             if module.padding_idx is not None:
                 module.weight.data[module.padding_idx].zero_()
 
-    # Adapted from transformers.modeling_utils.PreTrainedModel._check_and_enable_sdpa
+    # Adapted from transformers.modeling_utils.PreTrainedModel._autoset_attn_implementation
     @classmethod
-    def _check_and_enable_sdpa(cls, config, hard_check_only: bool = False) -> PretrainedConfig:
-        # We remove the checks on `is_torch_sdpa_available()` and `cls._supports_sdpa` as Falcon supports SDPA from torch==2.0.0 (no requirement on 2.1).
-        _is_bettertransformer = getattr(cls, "use_bettertransformer", False)
-        if _is_bettertransformer:
-            return config
-
-        if not hard_check_only:
-            config._attn_implementation = "sdpa"
+    def _autoset_attn_implementation(
+        cls,
+        config,
+        use_flash_attention_2: bool = False,
```

Comment:
overwrite to set "sdpa" no matter what because that is the only attention that IDEFICS has, and basically `attn_implementation` is not used anywhere except for when preparing the causal mask